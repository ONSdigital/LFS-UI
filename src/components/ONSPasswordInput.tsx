import React, { Component } from 'react';

interface Props{
  label?: string,
  id?: string,
  placeholder? : string
}

export class ONSPasswordInput extends Component <Props, {}>{
  constructor(props : Props) {
    super(props);
  }

    render() {
    return (
      <p className="field">
        {this.props.label !== undefined &&
        <label className="label">{this.props.label}</label>
        }
        <input type="password" id={this.props.id} className="input input--text input-type__input " placeholder={this.props.placeholder} />
      </p>
    );
  }
}
