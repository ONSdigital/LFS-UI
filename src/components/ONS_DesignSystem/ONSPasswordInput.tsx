import React, {ChangeEvent, Component} from "react";

interface Props {
    label?: string
    placeholder?: string
    marginTop?: number
    onChange?: (e: ChangeEvent<HTMLInputElement>, ...args: any[]) => void
    value: string
}

interface State {
    password: boolean
}

export class ONSPasswordInput extends Component <Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {password: true};
    }

    togglePassword = () => {
        this.setState({password: !this.state.password})
    };

    handleChange = (e: ChangeEvent<HTMLInputElement>) => {
        if (this.props.onChange !== undefined) {
            this.props.onChange(e, this.props.label);
        }
    };

    spacing = () => {
        let buttonStyle = {
            marginTop: String(this.props.marginTop) + "px",
          };
        return buttonStyle
    }


    render() {
        return (
            <p className="field">
                <label className="label" htmlFor="password">Password</label>
                <label className="label" htmlFor="password"/>
                <input
                    type={this.state.password ? "password" : "text"} id="password"
                    className="input input--text input-type__input u-mt-xs"
                    value={this.props.value}
                    onChange={(e) => this.handleChange(e)}
                />
                <span className="checkbox checkbox--toggle" style={this.spacing()}>
                    <input
                        type="checkbox"
                        id="password-toggle"
                        className="checkbox__input"
                        name="show-password"
                        onClick={this.togglePassword}
                    />
                    <label id="password-toggle-label" className="checkbox__label " htmlFor="password-toggle">
                        Show password
                    </label>
                </span>
            </p>

        );
    }
}
