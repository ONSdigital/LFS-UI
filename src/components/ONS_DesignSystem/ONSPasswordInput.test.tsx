import React from 'react';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { ONSPasswordInput } from './ONSPasswordInput';

describe("ONS Password Input Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {}

    const labelProps = {
        label: "Submit"
    }

    const changeProps = {
        onChange: jest.fn(),
    }

    const undefinedChangeProps = {
        onChange: undefined
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSPasswordInput 
                label={props.label}
                placeholder={props.placeholder}
                marginTop={props.marginTop}
                onChange={props.onChange}>
            </ONSPasswordInput>
        )
    }

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("should render with the correct label", () => {
        expect(wrapper(mount, labelProps).find("ONSPasswordInput").getElement().props.label).toEqual(labelProps.label);
    });

    it("should handle a change", () => {
        //defined onchange
        wrapper(mount, changeProps).find('input.input').simulate('change')
        expect(changeProps.onChange).toHaveBeenCalled()        
        
        //undefined onchange
        wrapper(mount, undefinedChangeProps).find('input.input').simulate('change')
        expect(undefinedChangeProps.onChange).toBeUndefined()        

    })

    it("should handle a click on the checkbox", () => {
        let thisWrapper = wrapper(mount, undefinedChangeProps)
        thisWrapper.find('input.checkbox__input').simulate('click')
        expect(thisWrapper.state('password')).toBeFalsy();
    })

})