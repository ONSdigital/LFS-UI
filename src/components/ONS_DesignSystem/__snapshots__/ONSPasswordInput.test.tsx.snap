// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Password Input Test matches Snapshot 1`] = `ShallowWrapper {}`;
